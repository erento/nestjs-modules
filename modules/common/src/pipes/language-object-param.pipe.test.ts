import {LanguageObjectParamPipe} from './language-object-param.pipe';

describe('LanguageObjectParamPipe', (): void => {
    let pipe: LanguageObjectParamPipe;

    beforeEach((): void => {
        pipe = new LanguageObjectParamPipe();
    });

    test.each<[any]>([
        [undefined],
        [null],
        [{}],
        [{lang: 'de-DE'}],
        [{lang: 'de-CH'}],
        [{lang: 'de-AT'}],
        [{lang: 'en-GB'}],
        [{lang: 'en-US'}],
        [{lang: 'es-ES'}],
        [{lang: 'fr-FR'}],
        [{lang: 'it-IT'}],
        [{lang: 'de'}],
        [{lang: 'en'}],
        [{lang: 'es'}],
        [{lang: 'fr'}],
        [{lang: 'it'}],
        [{lang: 'abc'}],
    ])('should transform query params properly', (input: any): void => {
        expect(pipe.transform(input)).toMatchSnapshot();
    });
});
