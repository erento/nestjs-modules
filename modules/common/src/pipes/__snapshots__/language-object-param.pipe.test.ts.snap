// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageObjectParamPipe should transform query params properly 1`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 2`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 3`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 4`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "de-DE",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 5`] = `
Object {
  "culture": "CH",
  "full": "de-CH",
  "locale": "de",
  "original": "de-CH",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 6`] = `
Object {
  "culture": "AT",
  "full": "de-AT",
  "locale": "de",
  "original": "de-AT",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 7`] = `
Object {
  "culture": "GB",
  "full": "en-GB",
  "locale": "en",
  "original": "en-GB",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 8`] = `
Object {
  "culture": "US",
  "full": "en-US",
  "locale": "en",
  "original": "en-US",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 9`] = `
Object {
  "culture": "ES",
  "full": "es-ES",
  "locale": "es",
  "original": "es-ES",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 10`] = `
Object {
  "culture": "FR",
  "full": "fr-FR",
  "locale": "fr",
  "original": "fr-FR",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 11`] = `
Object {
  "culture": "IT",
  "full": "it-IT",
  "locale": "it",
  "original": "it-IT",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 12`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "de",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 13`] = `
Object {
  "culture": "GB",
  "full": "en-GB",
  "locale": "en",
  "original": "en",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 14`] = `
Object {
  "culture": "ES",
  "full": "es-ES",
  "locale": "es",
  "original": "es",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 15`] = `
Object {
  "culture": "FR",
  "full": "fr-FR",
  "locale": "fr",
  "original": "fr",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 16`] = `
Object {
  "culture": "IT",
  "full": "it-IT",
  "locale": "it",
  "original": "it",
}
`;

exports[`LanguageObjectParamPipe should transform query params properly 17`] = `
Object {
  "culture": "DE",
  "full": "de-DE",
  "locale": "de",
  "original": "abc",
}
`;
